File: backend/src/main/kotlin/top/bearingwall/asya/model/DelegateAttrRecord.kt
package top.bearingwall.asya.model

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "delegate_attr_records")
class DelegateAttrRecord(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conference_id", nullable = false)
    var conference: Conference,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "delegate_id", nullable = false)
    var delegate: User,

    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "updated_at", nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "created_by")
    var createdBy: UUID? = null,

    @Column(name = "updated_by")
    var updatedBy: UUID? = null,

    @OneToMany(mappedBy = "record", cascade = [CascadeType.ALL], orphanRemoval = true)
    var attrValues: MutableSet<DelegateAttrValue> = mutableSetOf()
)

File: backend-dotnet/src/AsyaMun.Api/Models/CustomAttr/DelegateAttrRecord.cs
namespace AsyaMun.Api.Models;

public class DelegateAttrRecord
{
    public Guid Id { get; set; }

    public DateTime CreatedAt { get; set; }

    public Guid? CreatedBy { get; set; }

    public DateTime UpdatedAt { get; set; }

    public Guid? UpdatedBy { get; set; }

    public Guid ConferenceId { get; set; }

    public Guid DelegateId { get; set; }

    public virtual Conference Conference { get; set; } = null!;

    public virtual User Delegate { get; set; } = null!;

    public virtual ICollection<DelegateAttrValue> DelegateAttrValues { get; set; } = new List<DelegateAttrValue>();
}